src/code/cutscene_camera.c

```diff
@@ -353,28 +357,578 @@ void CutsceneCamera_Reset(void) {
+    } else {
+        waypoints[2] = interpState->duration - 1;
+    }
+
+    if ((interpState->waypoint + 3) < (interpState->duration - 1)) {
+        waypoints[3] = interpState->waypoint + 3;
+    } else {
+        waypoints[3] = interpState->duration - 1;
+    }
+
+    func_801629BC(new_var, coeff);
+
+    if (camPos != NULL) {
+        camPos->x = (coeff[0] * pointCmd[waypoints[0]].pos.x) + (coeff[1] * pointCmd[waypoints[1]].pos.x) +
+                    (coeff[2] * pointCmd[waypoints[2]].pos.x) + (coeff[3] * pointCmd[waypoints[3]].pos.x);
+        camPos->y = (coeff[0] * pointCmd[waypoints[0]].pos.y) + (coeff[1] * pointCmd[waypoints[1]].pos.y) +
+                    (coeff[2] * pointCmd[waypoints[2]].pos.y) + (coeff[3] * pointCmd[waypoints[3]].pos.y);
+        camPos->z = (coeff[0] * pointCmd[waypoints[0]].pos.z) + (coeff[1] * pointCmd[waypoints[1]].pos.z) +
+                    (coeff[2] * pointCmd[waypoints[2]].pos.z) + (coeff[3] * pointCmd[waypoints[3]].pos.z);
+    }
+
+    if (camFov != NULL) {
+        *camFov = (coeff[0] * miscCmd[waypoints[0]].fov) + (coeff[1] * miscCmd[waypoints[1]].fov) +
+                  (coeff[2] * miscCmd[waypoints[2]].fov) + (coeff[3] * miscCmd[waypoints[3]].fov);
+    }
+
+    if (camRoll != NULL) {
+        s16 targetRolls[4];
+        s32 sp2C[3];
+        s32 rollDiffToTarget;
+
+        targetRolls[0] = CAM_DEG_TO_BINANG(miscCmd[waypoints[0]].roll);
+        targetRolls[1] = CAM_DEG_TO_BINANG(miscCmd[waypoints[1]].roll);
+        targetRolls[2] = CAM_DEG_TO_BINANG(miscCmd[waypoints[2]].roll);
+        targetRolls[3] = CAM_DEG_TO_BINANG(miscCmd[waypoints[3]].roll);
+
+        sp2C[0] = (s16)(targetRolls[1] - targetRolls[0]);
+        sp2C[1] = sp2C[0] + (s16)(targetRolls[2] - targetRolls[1]);
+        sp2C[2] = sp2C[1] + (s16)(targetRolls[3] - targetRolls[2]);
+
+        rollDiffToTarget = ((coeff[1] * sp2C[0]) + (coeff[2] * sp2C[1]) + (coeff[3] * sp2C[2]));
+        *camRoll = targetRolls[0] + rollDiffToTarget;
+    }
+
+    interpState->curFrame++;
+
+    if (interpState->curFrame == pointCmd[interpState->waypoint + 1].duration) {
+        interpState->curFrame = 0;
+        interpState->waypoint++;
+        if (interpState->waypoint >= (interpState->duration - 3)) {
+            interpState->type = CS_CAM_INTERP_OFF;
+            return interpState->duration;
+        }
+    }
+    return 0;
+}
+
+static f32 sKnots[38];
+
+// Only used by unused CutsceneCamera_Interp_Unused
+void func_80162FF8(s16 arg0) {
+    f32 val = 0.0f;
+    s32 i;
+
+    sKnots[0] = 0.0f;
+    sKnots[1] = 0.0f;
+    sKnots[2] = 0.0f;
+
+    for (i = 3; i < arg0; i++) {
+        if (i == 3) {
+            val += 0.9f;
+        } else if ((i == 4) || (i == (arg0 - 1))) {
+            val += 0.6f;
+        } else {
+            val += 0.3f;
+        }
+
+        sKnots[i] = val;
+    }
+
+    val += 0.9f;
+    sKnots[i++] = val;
+    sKnots[i++] = val;
+    sKnots[i++] = val;
+}
+
+#define FUNC_801631DC_PATHING_ORDER 3
```

Comment:
```suggestion
#define FUNC_801631DC_ORDER 3
```
This isn't necessarily pathing related right?